Hotels/Controllers/HotelsController.cs

```diff
@@ -88,5 +88,29 @@ public async Task<ActionResult<Hotel>> DeleteHotel(long id)
 
             return hotel;
         }
+
+        // GET: api/Hotels/5/Rooms
+        [HttpGet("{id}/Rooms")]
+        public async Task<ActionResult<List<HotelRoom>>> GetHotelRooms(long hotelId)
+        {
+            var hotel = await repository.GetHotelRoomsById(hotelId);
+            return hotel.HotelRooms;
+        }
+
+        // POST: api/Hotels/5/Rooms/12
+        [HttpPost("{hotelId}/Rooms/{roomId}")]
+        public async Task<ActionResult> AddRoom(long hotelId, long roomId)
+        {
+            await repository.AddRoomAsync(hotelId, roomId);
+            return CreatedAtAction(nameof(AddRoom), new { hotelId, roomId }, null);
+        }
+
+        // DELETE: api/Hotels/5/Rooms/12
+        [HttpDelete("{hotelId}/Rooms/{roomId}")]
+        public async Task<ActionResult> DeleteRoom(long hotelId, long roomId)
```

Comment:
This route should receive `roomNumber` not `roomId`
```suggestion
        [HttpDelete("{hotelId}/Rooms/{roomNumber}")]
        public async Task<ActionResult> DeleteRoom(long hotelId, long roomNumber)
```